compiler/src/dotty/tools/dotc/transform/MacroTransform.scala

```diff
@@ -13,6 +13,8 @@ abstract class MacroTransform extends Phase {
 
   import ast.tpd.*
 
+  override def isRunnable(using Context) = super.isRunnable && !ctx.usesBestEffortTasty
```

Comment:
This exact shape of `isRunnable` comes up everywhere. It looks like it should be pushed upward in `super.isRunnable`.

Reply:
It only has this form for Transform phases and the GenBCode. I feel like it would be more confusing to put this into `Phase`, which would imply that no phase should run in the case of using betasty, which would be incorrect (Semantic db genaration and pickler should still run). Of course if you think differently I can change this